ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/StreamedQueryResource.java

```diff
@@ -144,8 +146,11 @@ private Response handleStatement(
       return Errors.badRequest(String.format(
           "Statement type `%s' not supported for this resource",
           statement.getClass().getName()));
+    } catch (final KsqlAuthorizationException e) {
+      return Errors.forbiddenKafkaAccess(e);
```

Comment:
This is gonna be interesting. The `KsqlAuthorizationException` might be thrown by other access denies errors, such as SR authorization. Or what if we extend the `KsqlAuthorizationException` to not authorize the execution of a specific command in the future?

How can you return the right error code here?

Reply:
Updated the AuthorizationException to a more specific exception
https://github.com/confluentinc/ksql/commit/2c50ac55fe33ae9c664f387e3971636c76486360